src/vmm/src/gdb/target.rs

```diff
@@ -0,0 +1,746 @@
+            regs.regs[0] = cpu_regs.rax;
+            regs.regs[1] = cpu_regs.rbx;
+            regs.regs[2] = cpu_regs.rcx;
+            regs.regs[3] = cpu_regs.rdx;
+            regs.regs[4] = cpu_regs.rsi;
+            regs.regs[5] = cpu_regs.rdi;
+            regs.regs[6] = cpu_regs.rbp;
+            regs.regs[7] = cpu_regs.rsp;
+
+            regs.regs[8] = cpu_regs.r8;
+            regs.regs[9] = cpu_regs.r9;
+            regs.regs[10] = cpu_regs.r10;
+            regs.regs[11] = cpu_regs.r11;
+            regs.regs[12] = cpu_regs.r12;
+            regs.regs[13] = cpu_regs.r13;
+            regs.regs[14] = cpu_regs.r14;
+            regs.regs[15] = cpu_regs.r15;
+
+            regs.rip = cpu_regs.rip;
+            regs.eflags = u32::try_from(cpu_regs.rflags).expect("Error converting rflags");
+
+            Ok(())
+        } else {
+            error!("Failed to read cpu registers");
+            Err(TargetError::NonFatal)
+        }
+    }
+
+    fn write_registers(&mut self, regs: &CoreRegs, tid: Tid) -> TargetResult<(), Self> {
+        let new_regs = kvm_regs {
+            rax: regs.regs[0],
+            rbx: regs.regs[1],
+            rcx: regs.regs[2],
+            rdx: regs.regs[3],
+            rsi: regs.regs[4],
+            rdi: regs.regs[5],
+            rbp: regs.regs[6],
+            rsp: regs.regs[7],
+
+            r8: regs.regs[8],
+            r9: regs.regs[9],
+            r10: regs.regs[10],
+            r11: regs.regs[11],
+            r12: regs.regs[12],
+            r13: regs.regs[13],
+            r14: regs.regs[14],
+            r15: regs.regs[15],
+
+            rip: regs.rip,
+            rflags: regs.eflags as u64,
+        };
+
+        if self.set_regs(new_regs, tid).is_err() {
+            return Err(TargetError::NonFatal);
+        }
+
+        Ok(())
+    }
+
+    fn read_addrs(
+        &mut self,
+        start_addr: <Self::Arch as Arch>::Usize,
+        data: &mut [u8],
+        tid: Tid,
+    ) -> TargetResult<usize, Self> {
+        let vmm = &self.vmm.lock().expect("Error unlocking vmm");
+        let memory = vmm.guest_memory();
+
+        let len = data.len();
+
+        let mut total_read = 0_usize;
+        let start_addr = usize::try_from(start_addr).expect("Unable to convert addr to usize");
+
+        while total_read < len {
+            let gaddr = start_addr + total_read;
+            let paddr =
+                match self.translate_gva(&vmm.vcpus_handles[tid_to_cpuid(tid)], gaddr as u64) {
+                    Ok(paddr) => usize::try_from(paddr).expect("Unable to convert addr to usize"),
+                    Err(e) => {
+                        error!("Error {e:?} translating gva on read address: {start_addr:X}");
+                        return Err(TargetError::NonFatal);
+                    }
+                };
+
+            let psize = arch::PAGE_SIZE;
+            let read_len = std::cmp::min(len - total_read, psize - (paddr & (psize - 1)));
```

Comment:
Let's leave a comment what this does (compute how much space is left in the page after paddr, I think?)

Reply:
Added thank you